Composer/plugins/azurePublish/src/index.ts

```diff
@@ -34,281 +35,366 @@ interface PublishConfig {
+        project.settings.runtime.path
+      ) {
+        runtimeCodePath = project.settings.runtime.path;
+      }
+
+      // Prepare the temporary project
+      // this writes all the settings to the root settings/appsettings.json file
+      await this.init(botFiles, runtimeCodePath, resourcekey);
+
+      // Merge all the settings
+      // this combines the bot-wide settings, the environment specific settings, and 2 new fields needed for deployed bots
+      // these will be written to the appropriate settings file inside the appropriate runtime plugin.
+      const mergedSettings = mergeDeep(fullSettings, settings);
 
+      // Prepare parameters and then perform the actual deployment action
       const customizeConfiguration: CreateAndDeployResources = {
+        accessToken,
         subscriptionID,
         name,
         environment,
         hostname,
         luisResource,
         language,
       };
-
-      // append provision resource into file
-      // TODO: here is where we configure the template for the runtime, and should be parameterized when we
-      // implement interchangeable runtimes
-      const resourcePath = path.resolve(
-        this.getProjectFolder(resourcekey, DEFAULT_RUNTIME),
-        'appsettings.deployment.json'
-      );
-      const appSettings = await readJson(resourcePath);
-      await writeJson(
-        resourcePath,
-        { ...appSettings, ...settings },
-        {
-          spaces: 4,
-        }
+      await this.performDeploymentAction(
+        project,
+        mergedSettings,
+        runtime,
+        project.id,
+        profileName,
+        jobId,
+        resourcekey,
+        customizeConfiguration
       );
+    };
 
-      this.azDeployer = new BotProjectDeploy({
-        subId: subscriptionID,
-        logger: (msg: any) => {
-          console.log(msg);
-          this.logMessages.push(JSON.stringify(msg, null, 2));
-        },
-        accessToken: accessToken,
-        projPath: this.getProjectFolder(resourcekey, 'azurewebapp'),
-      });
+    /**************************************************************************************************
+     * plugin methods
+     *************************************************************************************************/
+    publish = async (config: PublishConfig, project: IBotProject, metadata, user) => {
+      const {
+        // these are provided by Composer
+        profileName, // the name of the publishing profile "My Azure Prod Slot"
+
+        // these are specific to the azure publish profile shape
+        name,
+        environment,
+        settings,
+        accessToken,
+      } = config;
 
+      // get the bot id from the project
+      const botId = project.id;
+
+      // generate an id to track this deploy
+      const jobId = uuid();
+
+      // resource key to map to one provision resource
+      const resourcekey = md5([project.name, name, environment, settings?.MicrosoftAppPassword].join());
```

Comment:
it is probably not a great idea to use the app password this way even though we are hashing it.  should get a unique enough name from the project name, name, environment combo

Reply:
done